eventsocket/eventsocket.go

```diff
@@ -305,17 +305,36 @@ func (h *Connection) ReadEvent() ([]string, error) {
 // It's used after parsing plain text event headers, but not JSON.
 func copyHeaders(src *textproto.MIMEHeader, dst []string, decode bool) {
 	var err error
+	var customHeaderValues string
+	clen := len(customHeaderPrefix)
 	for k, v := range *src {
 		k = capitalize(k)
-		if decode && MapKeyIndex[k] != 0 {
-			dst[MapKeyIndex[k]], err = url.QueryUnescape(v[0])
-			if err != nil {
+		if decode {
+			if MapKeyIndex[k] != 0 {
+				dst[MapKeyIndex[k]], err = url.QueryUnescape(v[0])
+				if err != nil {
+					dst[MapKeyIndex[k]] = v[0]
+				}
+			} else if len(k) >= clen && k[0:clen] == customHeaderPrefix {
+				var headerValue string
+				headerValue, err = url.QueryUnescape(v[0])
+				if err != nil {
+					headerValue = v[0]
+				}
+				customHeaderValues = customHeaderValues + k + ":" + headerValue + "|"
+			}
+		} else {
+			if MapKeyIndex[k] != 0 {
 				dst[MapKeyIndex[k]] = v[0]
+			} else if len(k) >= clen && k[0:clen] == customHeaderPrefix {
+				customHeaderValues = customHeaderValues + k + ":" + v[0] + "|"
 			}
-		} else if MapKeyIndex[k] != 0 {
-			dst[MapKeyIndex[k]] = v[0]
 		}
 	}
+
+	if customHeaderValues != "" {
+		dst[CustomHeaders] = customHeaderValues
+	}
 }
```

Comment:
After you made those changes, i feel this is even cleaner, if it works
```
func copyHeaders(src *textproto.MIMEHeader, dst []string, decode bool) {
	var customHeaderValues string
	clen := len(customHeaderPrefix)
	for k, v := range *src {
		k = capitalize(k)
		if decode {
			escapedHeader, err := url.QueryUnescape(v[0])
			if err == nil {
				v[0] = escapedHeader
			}
		}

		if MapKeyIndex[k] != 0 {
			dst[MapKeyIndex[k]] = v[0]
		} else if len(k) >= clen && k[0:clen] == customHeaderPrefix {
			customHeaderValues = customHeaderValues + k + ":" + v[0] + "|"
		}
	}

	if customHeaderValues != "" {
		dst[CustomHeaders] = customHeaderValues
	}
}
```

Reply:
yes, i will execute this code and will update here, it is working or  not.

Reply:
yes it is working